admin/nix/flake.nix

```diff
@@ -0,0 +1,143 @@
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+{
+  description = "A flake for the Nextcloud desktop client";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    with flake-utils.lib;
+    eachSystem [ "aarch64-linux" "x86_64-linux" "aarch64-darwin" "x86_64-darwin" ] (system:
+        let
+          pkgs = import nixpkgs {
+            inherit system;
+          };
+
+          inherit (pkgs.lib.lists) optional optionals;
+          inherit (pkgs.lib.strings) hasPrefix optionalString;
+          isARM = hasPrefix "aarch64" system;
+
+          buildMacOSSymlinks = pkgs.runCommand "nextcloud-build-symlinks" {} ''
+            mkdir -p $out/bin
+            ln -s /usr/bin/xcrun /usr/bin/xcodebuild /usr/bin/iconutil $out/bin
+          '';
+
+          nativeBuildInputs = with pkgs; [
+            cmake
+            extra-cmake-modules
+            pkg-config
+            inkscape
+            qt5.wrapQtAppsHook
+          ] ++ optionals stdenv.isDarwin [
+            buildMacOSSymlinks
+          ];
+
+          buildInputs = with pkgs; [
+            sqlite
+            openssl
+            pcre
+
+            qt5.qtbase
+            qt5.qtquickcontrols2
+            qt5.qtsvg
+            qt5.qtgraphicaleffects
+            qt5.qtdeclarative
+            qt5.qttools
+            qt5.qtwebsockets
+
+            libsForQt5.karchive
+            libsForQt5.qtkeychain
+          ] ++ optionals (!isARM) [
+            # Qt WebEngine not available on ARM
+            qt5.qtwebengine
+          ] ++ optionals stdenv.isLinux [
+            inotify-tools
+            libcloudproviders
+            libsecret
+
+            libsForQt5.breeze-icons
+            libsForQt5.qqc2-desktop-style
+            libsForQt5.kio
+          ] ++ optionals stdenv.isDarwin [
+            libsForQt5.qt5.qtmacextras
+
+            darwin.apple_sdk.frameworks.UserNotifications
+          ];
+
+          packages.default = with pkgs; stdenv.mkDerivation rec {
+            inherit nativeBuildInputs buildInputs;
+            pname = "nextcloud-client";
+            version = "dev";
+            src = ../../.;
+
+            dontStrip = true;
+            enableDebugging = true;
+            separateDebugInfo = false;
+            enableParallelBuilding = true;
+
+            preConfigure = optionals stdenv.isLinux [
```

Comment:
preConfigure uses a multiline string and this only works because nix mangles lists to strings. In the future this might stop working.